pywren_ibm_cloud/job/job.py

```diff
@@ -184,56 +171,44 @@ def _create_job(config, internal_storage, executor_id, job_id, func, data, runti
         inc_modules = None
 
     logger.debug('ExecutorID {} | JobID {} - Serializing function and data'.format(executor_id, job_id))
-    # pickle func and all data (to capture module dependencies)
-
     serializer = SerializeIndependent(runtime_meta['preinstalls'])
     func_and_data_ser, mod_paths = serializer([func] + data, inc_modules, exc_modules)
-
-    func_str = func_and_data_ser[0]
     data_strs = func_and_data_ser[1:]
     data_size_bytes = sum(len(x) for x in data_strs)
+    module_data = create_module_data(mod_paths)
+    func_str = func_and_data_ser[0]
+    func_module_str = pickle.dumps({'func': func_str, 'module_data': module_data}, -1)
+    func_module_size_bytes = len(func_module_str)
+    total_size = utils.sizeof_fmt(data_size_bytes+func_module_size_bytes)
 
-    host_job_meta['agg_data'] = False
     host_job_meta['data_size_bytes'] = data_size_bytes
+    host_job_meta['func_module_size_bytes'] = func_module_size_bytes
 
-    log_msg = 'ExecutorID {} | JobID {} - Uploading function and data'.format(executor_id, job_id)
-    logger.info(log_msg)
-    if not log_level:
-        print(log_msg, end=' ')
-
-    if data_size_bytes < MAX_AGG_DATA_SIZE:
-        agg_data_key = create_agg_data_key(JOBS_PREFIX, executor_id, job_id)
-        job_description['data_key'] = agg_data_key
-        agg_data_bytes, agg_data_ranges = _agg_data(data_strs)
-        job_description['data_ranges'] = agg_data_ranges
-        agg_upload_time = time.time()
-        internal_storage.put_data(agg_data_key, agg_data_bytes)
-        host_job_meta['agg_data'] = True
-        host_job_meta['data_upload_time'] = time.time() - agg_upload_time
-        host_job_meta['data_upload_timestamp'] = time.time()
-    else:
-        log_msg = ('ExecutorID {} | JobID {} - Total data exceeded '
-                   'maximum size of {} bytes'.format(executor_id, job_id, MAX_AGG_DATA_SIZE))
+    if data_size_bytes > MAX_AGG_DATA_SIZE:
+        log_msg = ('ExecutorID {} | JobID {} - Total data exceeded maximum size '
+                   'of {} bytes'.format(executor_id, job_id, MAX_AGG_DATA_SIZE))
         raise Exception(log_msg)
 
-    module_data = create_module_data(mod_paths)
-    # Create func and upload
-    host_job_meta['func_name'] = func_name
```

Comment:
@JosepSampe I have seen that "func_name" was removed from future's status, is it ok to revert it back for "_call_status" dict? I need it for making PyWren reports efficiently

Reply:
ok, will recover it

Reply:
Added fix in #260 
You can access it with: `f.function_name`